integration/walletextension/wallet_extension_test.go

```diff
@@ -537,46 +533,59 @@ func createObscuroNetwork(t *testing.T) {
 
 	// Deploy an ERC20 contract to the Obscuro network.
 	txWallet := wallets.Tokens[bridge.OBX].L2Owner
+	generateAndSubmitViewingKey(walletExtensionAddr, walletExtensionAddr, txWallet.PrivateKey())
 	deployContractTx := types.LegacyTx{
 		Nonce:    simulation.NextNonce(clients, txWallet),
 		Gas:      1025_000_000,
 		GasPrice: common.Big0,
 		Data:     erc20contract.L2BytecodeWithDefaultSupply("TST"),
 	}
-	generateAndSubmitViewingKey(t, walletExtensionAddr, walletExtensionAddr, txWallet.PrivateKey())
-	txBinaryHex, err := formatTxForSubmission(txWallet, &deployContractTx)
+	sendTransactionAndAwaitConfirmation(txWallet, deployContractTx)
+}
+
+// Generates a new account and registers it with the node.
+func registerPrivateKey(t *testing.T) (common.Address, *ecdsa.PrivateKey) {
+	privateKey, err := crypto.GenerateKey()
+	if err != nil {
+		t.Fatal(err)
+	}
+	accountAddr := crypto.PubkeyToAddress(privateKey.PublicKey)
+	generateAndSubmitViewingKey(walletExtensionAddr, accountAddr.String(), privateKey)
+	return accountAddr, privateKey
+}
+
+// Submits a transaction and awaits the transaction receipt.
+func sendTransactionAndAwaitConfirmation(txWallet wallet.Wallet, deployContractTx types.LegacyTx) map[string]interface{} {
+	txBinaryHex, err := signAndSerialiseTransaction(txWallet, &deployContractTx)
 	if err != nil {
 		panic(fmt.Sprintf("failed to create test Obscuro network. Cause: %s", err))
 	}
 
-	sendTxJSON := makeEthJSONReqAsJSON(t, walletExtensionAddr, rpcclientlib.RPCSendRawTransaction, []interface{}{txBinaryHex})
+	sendTxJSON := makeEthJSONReqAsJSON(walletExtensionAddr, rpcclientlib.RPCSendRawTransaction, []interface{}{txBinaryHex})
 
-	// Verify the Obscuro ERC20 contract deployed successfully
-	_, ok := sendTxJSON[walletextension.RespJSONKeyResult].(string)
+	// Verify the transaction was successful.
+	txHash, ok := sendTxJSON[walletextension.RespJSONKeyResult].(string)
 	if !ok {
 		panic("could not retrieve transaction hash from JSON result, failed to deploy ERC20")
 	}
 
-	// We wait ten seconds for the first rollup to be published, to ensure the network is ready.
-	firstRollupIdx := 1
 	counter := 0
 	for {
 		if counter > 10 {
-			t.Fatalf("first rollup had not been published after 10 seconds")
+			panic("could not get ERC20 receipt after 10 seconds")
```

Comment:
Is 10 seconds necessary? There's 12 tests in here, in worst case scenario does this add 2mins to the time for build to fail?

Reply:
No, I think you'd fail after the first test (10 secs). Plus I'm not worried about the tests being slow when they're broken. It's them being slow when they pass that worries me more.